src/plugins/unified_field_list/public/components/field_visualize_button/field_visualize_button.tsx

```diff
@@ -7,29 +7,48 @@
  */
 
 import React, { useEffect, useState } from 'react';
+import { EuiButtonProps } from '@elastic/eui';
 import { METRIC_TYPE, UiCounterMetricType } from '@kbn/analytics';
 import type { DataView, DataViewField } from '@kbn/data-views-plugin/public';
-import { triggerVisualizeActions, VisualizeInformation } from './lib/visualize_trigger_utils';
-import { getVisualizeInformation } from './lib/visualize_trigger_utils';
-import { DiscoverFieldVisualizeInner } from './discover_field_visualize_inner';
+import type { UiActionsStart } from '@kbn/ui-actions-plugin/public';
+import { FieldVisualizeButtonInner } from './field_visualize_button_inner';
+import {
+  triggerVisualizeActions,
+  getVisualizeInformation,
+  type VisualizeInformation,
+} from './visualize_trigger_utils';
 
-interface Props {
+export interface FieldVisualizeButtonProps {
   field: DataViewField;
   dataView: DataView;
+  originatingApp: string; // plugin id
+  uiActions: UiActionsStart;
   multiFields?: DataViewField[];
-  contextualFields: string[];
+  contextualFields?: string[]; // names of fields which were also selected (like columns in Discover grid)
   trackUiMetric?: (metricType: UiCounterMetricType, eventName: string | string[]) => void;
+  buttonProps?: Partial<EuiButtonProps>;
+  wrapInContainer?: (element: React.ReactElement) => React.ReactElement;
 }
 
-export const DiscoverFieldVisualize: React.FC<Props> = React.memo(
-  ({ field, dataView, contextualFields, trackUiMetric, multiFields }) => {
+export const FieldVisualizeButton: React.FC<FieldVisualizeButtonProps> = React.memo(
```

Comment:
Hi @stratoula ! In this PR I am moving "Visualize" button and the related code to unifiedFieldList plugin. And I am wondering if we could also move "Explore in Discover" button here as well. I guess I would then have to convert it into an action and trigger in the same way as we do for "Visualize" button.  Would it make sense to do so?

Reply:
Hmm, yes we should def move it on the unified plugin. The reasons we used href here(using the discover locator here) are:
- it is in sync with the same link that exists on the topNav of Lens
- we open it on a new tab

Can we keep the same functionality here without adding a new trigger action?

Reply:
@stratoula Yes, with a trigger we could also open in a new tab and it could be similar to [lens/public/trigger_actions/open_in_discover_action.ts](https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/public/trigger_actions/open_in_discover_action.ts) and [unified_field_list/public/components/field_visualize_button/field_visualize_button_inner.tsx](https://github.com/jughosta/kibana/blob/a66193049cf1b7bed49874eafc6c16eca073bd1e/src/plugins/unified_field_list/public/components/field_visualize_button/field_visualize_button_inner.tsx#L36) (at least from what I understand about triggers). 
On the other hand, a direct call to discover locator without a trigger would be problematic from unifiedFieldList unless we add it as a dependency.

Reply:
Ok then, if we can have the same functionality let's do it with a uiAction instead :)

Reply:
@jughosta it could also be a prop right? I mean whichever app wants to use the unified field list and wants to display the button can give the props of the button (label, action etc) so it depends on the application. This could possibly make the field list more flexible (?).

Reply:
@stratoula Yes, that might work, thanks! Created a follow up ticket: https://github.com/elastic/kibana/issues/142361 
Anyway, the current implementation is also simple and good so it can definitely wait :)